auth/api/iam/api.go

```diff
@@ -80,8 +79,7 @@ func (r Wrapper) Routes(router core.EchoRouter) {
 				// Add http.Request to context, to allow reading URL query parameters
 				requestCtx := context.WithValue(ctx.Request().Context(), httpRequestContextKey, ctx.Request())
 				ctx.SetRequest(ctx.Request().WithContext(requestCtx))
-				// TODO: Do we need a generic error handler?
-				// ctx.Set(core.ErrorWriterContextKey, &protocolErrorWriter{})
+				ctx.Set(core.ErrorWriterContextKey, &oauth2ErrorWriter{})
```

Comment:
Must all APIs under `/aim` return oauth errors?

Reply:
For now I'd say yes, and it doesn't hurt. When new non-OAuth2 APIs are added to IAM, we could/should introduce a skipper func?

Reply:
I see we also have `/internal/auth/v2/{did}/request-access-token`. I'll only apply it for request URLs starting with `/iam/`